client/gutenberg/extensions/slideshow/slideshow.js

```diff
@@ -13,6 +13,33 @@ import createSwiper from './create-swiper';
 
 const SIXTEEN_BY_NINE = 16 / 9;
 
+/**
+ * Swiper event handler.
+ *
+ * A Swiper instance is passed as `this` to this function.
+ */
+function swiperInit() {
+	this.el.classList.add( 'wp-swiper-initialized' );
+	swiperAutoSize.bind( this )();
+}
+
+/**
+ * Swiper event handler.
+ *
+ * A Swiper instance is passed as `this` to this function.
+ */
+function swiperAutoSize() {
+	const img = this.el.querySelector( '.swiper-slide[data-swiper-slide-index="0"] img' );
+	if ( ! img ) {
+		return;
+	}
+	const aspectRatio = img.clientWidth / img.clientHeight;
+	const sanityAspectRatio = Math.max( Math.min( aspectRatio, SIXTEEN_BY_NINE ), 1 );
+	const sanityHeight = typeof window !== 'undefined' ? window.innerHeight * 0.8 : 600;
+	const swiperHeight = Math.min( this.width / sanityAspectRatio, sanityHeight );
+	this.el.style.height = `calc( ${ Math.floor( swiperHeight ) }px + 4em )`;
+}
+
```

Comment:
These could also be made `static` methods of the `Slideshow` class, but I thought that might be even more confusing with regard to the `this`?

Reply:
Maybe? Currently `view.js` doesn't import `Slideshow`, so we'd be complicating the import structure a bit. Also, to go back to the hypothetical case in which the callbacks needed access to `props` or `state`, how would that work? In other words, what would we gain from `static` methods that differs from extracting them from the Component's scope as you had it in your first pass?

Reply:
> In other words, what would we gain from `static` methods that differs from extracting them from the Component's scope as you had it in your first pass? 

Not much gain other than some (debatable) encapsulation.

Marking this as resolved since per DM, we're not going to follow this path anyway.